lib/java/src/main/java/org/apache/thrift/server/AbstractNonblockingServer.java

```diff
@@ -351,7 +359,13 @@ public boolean read() {
 
           // if this frame will push us over the memory limit, then return.
           // with luck, more memory will free up the next time around.
-          if (readBufferBytesAllocated.get() + frameSize > MAX_READ_BUFFER_BYTES) {
+          long currentAllocated = readBufferBytesAllocated.get();
```

Comment:
In case the method is overridden, should probably use the new method to get this instead.

```suggestion
          long currentAllocated = getReadBufferBytesAllocated();
```

Reply:
Alternatively, just make the method final, since it depends so much on the internal implementation.

Reply:
made it final which makes it safer to call the getter